src/services/codefixes/fixImportNonExportedMember.ts

```diff
@@ -0,0 +1,114 @@
+        },
+        fixIds: [fixId],
+        getAllCodeActions: context => codeFixAll(context, errorCodes, (changes, diag) => {
+            const info = getInfo(diag.file, context, diag.start);
+            if (info) doChange(changes, info.originSourceFile, info.node);
+        }),
+    });
+
+    interface Info {
+        readonly node: Identifier;
+        readonly importDecl: ImportDeclaration;
+        readonly originSourceFile: SourceFile
+    }
+    function isImportDeclaration(node: Node): node is ImportDeclaration {
+        return node.kind === SyntaxKind.ImportDeclaration;
+    }
+
+    function getInfo(sourceFile: SourceFile, context: CodeFixContext | CodeFixAllContext, pos: number): Info | undefined {
+        const node = getTokenAtPosition(sourceFile, pos);
+        if (node && isIdentifier(node)) {
+            const importDecl = findAncestor(node, isImportDeclaration);
+            if (!importDecl || !isStringLiteralLike(importDecl.moduleSpecifier)) {
+                return undefined;
+            }
+            const resolvedModule = getResolvedModule(sourceFile, importDecl.moduleSpecifier.text);
+            const originSourceFile = resolvedModule && context.program.getSourceFile(resolvedModule.resolvedFileName);
+            if (!originSourceFile) {
+                return undefined;
+            }
+            return { node, importDecl, originSourceFile };
+        }
+    }
+
+    function getNamedExportDeclaration(sourceFile: SourceFile): ExportDeclaration | undefined {
+        let namedExport;
+        const statements = sourceFile.statements.filter(isExportDeclaration);
+        for (const statement of statements) {
+            if (statement.exportClause && isNamedExports(statement.exportClause)) {
+                namedExport = statement;
+            }
+        }
+        return namedExport;
+    }
+
+    function compareIdentifiers(s1: Identifier, s2: Identifier) {
+        return compareStringsCaseInsensitive(s1.text, s2.text);
+    }
+
+    function sortSpecifiers(specifiers: ExportSpecifier[]): readonly ExportSpecifier[] {
+        return stableSort(specifiers, (s1, s2) => {
+            return compareIdentifiers(s1.propertyName || s1.name, s2.propertyName || s2.name) ||
+                compareIdentifiers(s1.name, s2.name);
+        });
+    }
+
+    function doChange(changes: textChanges.ChangeTracker, sourceFile: SourceFile, node: Identifier): void {
+        const moduleSymbol = sourceFile.localSymbol || sourceFile.symbol;
+        const localSymbol = moduleSymbol.valueDeclaration.locals?.get(node.escapedText);
+        if (!localSymbol) {
+            return;
+        }
+        if (isFunctionSymbol(localSymbol)) {
+            const start = localSymbol.valueDeclaration.pos;
+            changes.insertExportModifierAt(sourceFile, start ? start + 1 : 0);

```

Comment:
What exactly is with the `start + 1`? I don't understand.

Reply:
I want to add an `export ` identifier in front of non-exported type, you can see the `codeFixImportNonExportedMember1.ts`, an export added before `function bar()`, the start there is the position of line break, so need to +1 for start.

Reply:
The problem you are running into is more about using the `pos` (a.k.a. the "full start") instead of the start after the trivia (what you'd get when you call `getStart(sourceFile)`).

As far as I can tell, a test like

```ts
/**
 * hello
 */
function bar() {
}
```

will likely end up with the export inside of the comment..

What happens when you just use `insertExportModifier` instead of `insertExportModifierAt`?